crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -797,6 +827,46 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.types.definitions.insert(definition, context_expr_ty);
     }
 
+    /// Infer the types of exceptions caught by an exception handler
+    fn infer_handled_exception_types<'a, 'b>(
+        &'b self,
+        handled_exceptions: &'a ast::Expr,
+    ) -> ExceptionHandlerTypeIterator<'a, 'b, 'db> {
+        match handled_exceptions {
+            ast::Expr::Tuple(multiple_exceptions) => ExceptionHandlerTypeIterator::Multiple {
+                exceptions: multiple_exceptions.into_iter(),
+                inference_builder: self,
+            },
+            single_exception => ExceptionHandlerTypeIterator::Single {
+                exception: Some(single_exception),
+                inference_builder: self,
+            },
+        }
+    }
```

Comment:
We can simplify the lifetimes here because all AST nodes have the lifetime `'db`. 
```suggestion
    fn infer_handled_exception_types(
        &self,
        handled_exceptions: &'db ast::Expr,
    ) -> ExceptionHandlerTypeIterator<'_, 'db> {
        match handled_exceptions {
            ast::Expr::Tuple(multiple_exceptions) => ExceptionHandlerTypeIterator::Multiple {
                exceptions: multiple_exceptions.into_iter(),
                inference_builder: self,
            },
            single_exception => ExceptionHandlerTypeIterator::Single {
                exception: Some(single_exception),
                inference_builder: self,
            },
        }
    }
```